File: tests/fixtures/sample.ts
/** Sample TypeScript file for testing ast-grep and LSP integration. */

interface CalculatorOptions {
  initialValue?: number;
}

class Calculator {
  private value: number;

  constructor(options: CalculatorOptions = {}) {
    this.value = options.initialValue ?? 0;
  }

  add(n: number): this {
    this.value += n;
    return this;
  }

  multiply(n: number): this {
    this.value *= n;
    return this;
  }

  result(): number {
    return this.value;
  }

  toString(): string {
    return String(this.value);
  }
}

function greet(name: string): string {
  return `Hello, ${name}!`;
}

function divide(a: number, b: number): number | null {
  if (b === 0) return null;
  return a / b;
}

const calc = new Calculator({ initialValue: 0 });
calc.add(10).multiply(3);
console.log(`Result: ${calc}`);
console.log(greet("world"));

File: tests/fixtures/sample.py
"""Sample Python file for testing ast-grep and LSP integration."""

from typing import Optional


class Calculator:
    """A simple calculator for testing."""

    def __init__(self, value: float = 0.0) -> None:
        self.value = value

    def add(self, n: float) -> "Calculator":
        self.value += n
        return self

    def multiply(self, n: float) -> "Calculator":
        self.value *= n
        return self

    def result(self) -> float:
        return self.value

    def __str__(self) -> str:
        return str(self.value)


def greet(name: str) -> str:
    return f"Hello, {name}!"


def divide(a: float, b: float) -> Optional[float]:
    if b == 0:
        return None
    return a / b


if __name__ == "__main__":
    calc = Calculator(0.0)
    calc.add(10.0).multiply(3.0)
    print(f"Result: {calc}")
    print(greet("world"))
